extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -0,0 +1,381 @@
+      user().assert(target,
+        'amp-lightbox-viewer.open: element with id: %s not found', targetId);
+    }
+    return this.open_(target);
+  }
+
+  /**
+   * Opens the lightbox-viewer and displays the given element inside.
+   * @param {!Element} element Element to lightbox.
+   * @private
+   * @return {!Promise}
+   */
+  open_(element) {
+    if (this.activeElement_ == element) {
+      return;
+    }
+
+    const updateViewerPromise = this.updateViewer_(element);
+    this.getViewport().enterLightboxMode();
+
+    toggle(this.element, true);
+    this.active_ = true;
+
+    this.win.document.documentElement.addEventListener(
+        'keydown', this.boundHandleKeyboardEvents_);
+
+    return updateViewerPromise;
+  }
+
+  /**
+   * Closes the lightbox-viewer
+   * @private
+   */
+  close_() {
+    if (!this.active_) {
+      return;
+    }
+
+    toggle(this.element, false);
+    this.tearDownElement_(this.activeElement_);
+    this.getViewport().leaveLightboxMode();
+
+    this.activeElement_ = null;
+    this.active_ = false;
+
+    this.container_.setAttribute('no-prev', '');
+    this.container_.setAttribute('no-next', '');
+
+    this.win.document.documentElement.removeEventListener(
+        'keydown', this.boundHandleKeyboardEvents_);
+  }
+
+  /**
+   * Opens the next element to be displayed in the lightbox.
+   * @private
+   * @return {!Promise}
+   */
+  next_() {
+    dev().assert(this.activeElement_);
+    return this.manager_.getNext(this.activeElement_).then(nextElement => {
+      if (nextElement) {
+        return this.updateViewer_(nextElement);
+      }
+    });
+  }
+
+  /**
+   * Opens the previous element to be displayed in the lightbox.
+   * @private
+   * @return {!Promise}
+   */
+  previous_() {
+    dev().assert(this.activeElement_);
+    return this.manager_.getPrevious(this.activeElement_).then(prevElement => {
+      if (prevElement) {
+        return this.updateViewer_(prevElement);
+      }
+    });
+  }
+
+  /**
+   * Updates the viewer to display the new element and tears down the old one
+   * @param {!Element} newElement
+   * @private
+   * @return {!Promise}
+   */
+  updateViewer_(newElement) {
+    const previousElement = this.activeElement_;
+    dev().assert(newElement);
+    dev().assert(newElement != previousElement);
+
+    // tear down the previous element
+    if (previousElement) {
+      this.tearDownElement_(previousElement);
+    }
+
+    // setup the new element
+    this.setupElement_(newElement);
+
+    // update active element to be the new element
+    this.activeElement_ = newElement;
+
+    // update the controls
+    const updateControlsPromise = this.updateControls_();
+
+    // TODO(aghassemi): Preloading of +/- 1 elements
+
+    // TODO(aghassemi): This is a giant hack.
+    // Find a proper way of scheduling layout for a resource that does not
+    // not belong to the element requesting the layout.
+    if (newElement.resources_) {
```

Comment:
I'd still recommend to rewrite this like this:
```
if (newElement.__AMP__RESOURCE) {
  newElement.__AMP__RESOURCE.setInViewport(true);
  resourcesFor(this.win).scheduleLayout(newElement, newElement);
}
```

Otherwise it will simply not worked when compiled since `resources_` here and in core js will likely be obfuscated differently.